server/routes/categories.js

```diff
@@ -1,10 +1,191 @@
 const express = require('express');
 const router = express.Router();
+const Category = require('../db/Models/Category');
+const Post = require('../db/Models/Post');
 
 
-router.get('/smoke', (req, res) => {
-    res.send('smoke test on categories route');
-})
+router.get('/', (req, res) => {
+  return Category.fetchAll()
+    .then(cats => {
+      return res.json(cats);
+    })
+    .catch(err => {
+      return res.status(500).json({ message: err.message, code: err.code });
+    });
+});
 
 
+router.get('/automotive', (req, res) => {
+  return Post.fetchAll({
+    withRelated: ['categoryId', 'customerId', 'postStatusId', 'postPriorityId']
+  })
+    .then(posts => {
+      const results = posts.toJSON();
+      const automotive = results.filter(element => {
+        const elemCatId = element.category_id;
+        return elemCatId.id === 1;
+      });
+      return res.send(automotive);
```

Comment:
Probably want to use `res.json` here instead of `res.send`

This comments applies for all other categories as well.